src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -0,0 +1,152 @@
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.VS;
+using EncInterop = Microsoft.VisualStudio.LanguageServices.Implementation.EditAndContinue.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Managed.VS.EditAndContinue
+{
+
+    [Export(ExportContractNames.VsTypes.ProjectNodeComExtension)]
+    [AppliesTo(ProjectCapability.ManagedEditAndContinue)]
+    [ComServiceIid(typeof(IVsENCRebuildableProjectCfg))]
+    [ComServiceIid(typeof(EncInterop.IVsENCRebuildableProjectCfg2))]
+    [ComServiceIid(typeof(EncInterop.IVsENCRebuildableProjectCfg4))]
+    internal class EditAndContinueProvider : IVsENCRebuildableProjectCfg, EncInterop.IVsENCRebuildableProjectCfg2, EncInterop.IVsENCRebuildableProjectCfg4
+    {
+        private readonly ILanguageServiceHost _host;
+
+        [ImportingConstructor]
+        public EditAndContinueProvider(ILanguageServiceHost host)
+        {
+            _host = host;
+        }
+
+        #region IVsENCRebuildableProjectCfg Implementation
+        // Managed ENC always uses IVsENCRebuildableProjectCfg2.
+        // Although we don't implement the methods of IVsENCRebuildableProjectCfg, it is important that we implement the interface and return VSConstants.E_NOTIMPL.
+        // This is how the EncManager recognizes the project system that supports EnC.
+        public int ENCRebuild(object in_pProgram, out object out_ppSnapshot)
+        {
+            out_ppSnapshot = null;
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int BelongToProject(string in_szFileName, ENC_REASON in_ENCReason, int in_fOnContinue)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int ENCComplete(int in_fENCSuccess)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int CancelENC()
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int ENCRelink([In, MarshalAs(UnmanagedType.IUnknown)] object pENCRelinkInfo)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int StartDebugging()
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int StopDebugging()
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+
+        public int SetENCProjectBuildOption([In] ref Guid in_guidOption, [In, MarshalAs(UnmanagedType.LPWStr)] string in_szOptionValue)
+        {
+            return VSConstants.E_NOTIMPL;
+        }
+        #endregion
+
+        #region IVsENCRebuildableProjectCfg2 Implementation
+        public int BuildForEnc([In, MarshalAs(UnmanagedType.IUnknown)] object pUpdatePE)
+        {
+            return ((EncInterop.IVsENCRebuildableProjectCfg2)_host.HostSpecificEditAndContinueService)?.BuildForEnc(pUpdatePE) ?? VSConstants.S_OK;
```

Comment:
`AbstractProject`, the language service, when it does not have the `EditAndContinueImplOpt` object, to which the EnC questions are directed, returns a bunch of HR values. I have followed the same logic to return the same values when the language service is not available.

You can see the behavior of the `AbstractProject` [here](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/AbstractProject_IVsENCRebuildableProjectCfg.cs,22)